PiwikTracker/Dispatcher.swift

```diff
@@ -4,7 +4,9 @@ public protocol Dispatcher {
     
     var baseURL: URL { get }
     
-    var userAgent: String? { get }
+    var userAgent: String? { get set }
+    
+    func setUserAgent(ua: String)
```

Comment:
To me it feels like the user agent should be immutable once the dispatcher have been created. This will also leave the protocol unchanged.

Instead of making it mutable I suggest providing it as an optional parameter in the constructor in the protocol implementation.

This will make the code safer and more easy to reason about.